server/workflowarchive/archived_workflow_server.go

```diff
@@ -188,3 +191,28 @@ func (w *archivedWorkflowServer) ListArchivedWorkflowLabelValues(ctx context.Con
 	}
 	return labels, err
 }
+
+func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req *workflowarchivepkg.RetryArchivedWorkflowRequest) (*wfv1.Workflow, error) {
+	wfClient := auth.GetWfClient(ctx)
+	kubeClient := auth.GetKubeClient(ctx)
+
+	wf, err := w.wfArchive.GetWorkflow(req.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	wf, err = util.RetryWorkflow(ctx, kubeClient.CoreV1().Pods(wf.Namespace), wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), wf, req.RestartSuccessful, req.NodeFieldSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	wf, err = wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Create(ctx, wf, metav1.CreateOptions{})
+	if err != nil {
+		if apierr.IsAlreadyExists(err) {
+			return nil, fmt.Errorf("workflow already exists : %s", err)
```

Comment:
either:

* We should we ignore this error for idempotence? @sarabala1979 
* Do not wrap it .